dietpi/dietpi-software

```diff
@@ -8385,6 +8416,231 @@ _EOF_
+					else
+						invalid_text='[ERROR] No valid entry found, value needs to be a sequence of integers. Please retry...\n\n'
+					fi
+
+				done
+
+				if (( $obfs4port < 1024 )); then
+					G_EXEC setcap cap_net_bind_service=+ep "$(command -v obfs4proxy)"
+				fi
+
+				cat << _EOF_ > /etc/tor/torrc
+BridgeRelay 1
+
+ORPort $orport
+
+ServerTransportPlugin obfs4 exec $(command -v obfs4proxy)
+
+ServerTransportListenAddr obfs4 0.0.0.0:$obfs4port
+
+ExtORPort auto
+_EOF_
+
+				G_EXEC mkdir -p /etc/systemd/system/tor@.service.d
+				G_EXEC eval "echo -e '[Service]\nNoNewPrivileges=no' > /etc/systemd/system/tor@.service.d/dietpi-relay.conf"
+				G_EXEC systemctl daemon-reload
+
+			fi
+		
+			elif [[ $G_WHIP_RETURNED_VALUE == 'Guard/Middle' ]]; then
+		
+				cat << _EOF_ > /etc/tor/torrc
+ORPort $orport
+ExitRelay 0
+SocksPort 0
+ControlSocket 0
+_EOF_
+			elif [[ $G_WHIP_RETURNED_VALUE == 'Exit' ]]; then
+				G_AGI unbound
+				aSOFTWARE_INSTALL_STATE[182]=1
+
+				cat << _EOF_ > /etc/tor/torrc
+ORPort $orport
+ExitRelay 1
+SocksPort 0
+ControlSocket 0
+
+ExitPolicy accept *:20-21
+ExitPolicy accept *:43
+ExitPolicy accept *:53
+ExitPolicy accept *:79
+ExitPolicy accept *:80-81
+ExitPolicy accept *:88
+ExitPolicy accept *:110
+ExitPolicy accept *:143
+ExitPolicy accept *:220
+ExitPolicy accept *:389
+ExitPolicy accept *:443
+ExitPolicy accept *:464
+ExitPolicy accept *:531
+ExitPolicy accept *:543-544
+ExitPolicy accept *:554
+ExitPolicy accept *:636
+ExitPolicy accept *:706
+ExitPolicy accept *:749
+ExitPolicy accept *:873
+ExitPolicy accept *:902-904
+ExitPolicy accept *:981
+ExitPolicy accept *:989-990
+ExitPolicy accept *:991
+ExitPolicy accept *:992
+ExitPolicy accept *:993
+ExitPolicy accept *:995
+ExitPolicy accept *:1194
+ExitPolicy accept *:1220
+ExitPolicy accept *:1293
+ExitPolicy accept *:1500
+ExitPolicy accept *:1533
+ExitPolicy accept *:1677
+ExitPolicy accept *:1723
+ExitPolicy accept *:1755
+ExitPolicy accept *:1863
+ExitPolicy accept *:2082
+ExitPolicy accept *:2083
+ExitPolicy accept *:2086-2087
+ExitPolicy accept *:2095-2096
+ExitPolicy accept *:2102-2104
+ExitPolicy accept *:3690
+ExitPolicy accept *:4321
+ExitPolicy accept *:4643
+ExitPolicy accept *:5050
+ExitPolicy accept *:5190
+ExitPolicy accept *:5222-5223
+ExitPolicy accept *:5228
+ExitPolicy accept *:8008
+ExitPolicy accept *:8074
+ExitPolicy accept *:8082
+ExitPolicy accept *:8087-8088
+ExitPolicy accept *:8232-8233
+ExitPolicy accept *:8332-8333
+ExitPolicy accept *:8443
+ExitPolicy accept *:8888
+ExitPolicy accept *:9418
+ExitPolicy accept *:10000
+ExitPolicy accept *:11371
+ExitPolicy accept *:19294
+ExitPolicy accept *:19638
+ExitPolicy accept *:50002
+ExitPolicy accept *:64738
+ExitPolicy reject *:*
+_EOF_
+				if [[ ! ipv6 == '' ]]; then
+					echo "IPv6Exit 1" >> /etc/tor/torrc
+				fi
+				G_CONFIG_INJECT 'nameserver[[:blank:]]' 'nameserver 127.0.0.1' /etc/resolv.conf
+			fi
+
+			echo "ControlPort 9051" >> /etc/tor/torrc
+			if [[ ! $nickname == '' ]]; then
```

Comment:
Simpler check:
```
[[ $nickname ]]
```
has the meaning "is declared and not empty".

One liner:
```
[[ $nickname ]] && echo "Nickname $nickname" >> /etc/tor/torrc
```

Same below.